src/tasks.py

```diff
@@ -16,29 +16,131 @@
+    # an attribute is not found and NoneType is returned.
+    try:
+        # Retrieve summary related results.
+        section = report.add_section()
+        section.add_header("Run Summary")
+        section.add_bullet(
+                'Program: {0} - {1}'.format(
+                    check_xml_attrib(xml_file,'creator/program'),
+                    check_xml_attrib(xml_file,'creator/version')))
+        section.add_bullet(
+                'Command Line: {0}'.format(
+                    check_xml_attrib(
+                        xml_file,
+                        'creator/execution_environment/command_line')))
+        section.add_bullet(
+                'Start Time: {0}'.format(
+                    check_xml_attrib(
+                        xml_file,
+                        'creator/execution_environment/start_time')))
+        section.add_bullet(
+                f"Elapsed Time: {check_xml_attrib(xml_file, 'report/elapsed_seconds')}"
+            )
+
+        # Retrieve results from each of the scanner runs and display in table
+        feature_files = xml_file.find(".//feature_files")
+        scanner_results = []
+        section = report.add_section()
+        if feature_files is not None:
+            section.add_header('Scanner Results\n')
+            for name, count in zip(xml_file.findall(".//feature_file/name"),
+                                    xml_file.findall(".//feature_file/count")):
+                scanner_results.append({"Name": name.text, "Count": int(count.text)})
+                features_count += int(count.text)
+            sorted_scanner_results = sorted(
+                scanner_results, key=lambda x: x["Count"], reverse=True)
+            columns = scanner_results[0].keys()
+            t = MarkdownTable(columns)
+            for scanner_result in sorted_scanner_results:
+                print([str(scanner_result[column]) for column in columns])
+                t.add_row([str(scanner_result[column]) for column in columns])
+            section.add_table(t)
+        else:
+            section.add_header("There are no findings to report.")
+    except AttributeError as exception:
+        #log.warning(
+        #    f'Error parsing feature from Bulk Extractor report: {exception!s}')
+        raise exception
+    report.summary = f'{features_count} artifacts have been extracted.'
+    return report
+
+def extract_non_empty_files(artifact_dir, output_path) -> List[Dict]:
+    out_dir = os.path.join(artifact_dir, "output")
+    os.makedirs(out_dir, exist_ok=True)
+    out_files = []
+    for entry in glob.glob(os.path.join(artifact_dir, '**'), recursive=True):
+        if os.path.exists(entry) and not os.path.isdir(entry):
+            with open(entry, "rb") as f:
+                content = f.read()
+                if content:
+                    out_file = create_output_file(output_path, display_name=os.path.basename(entry))
+                    with open(out_file.path, "wb") as out_f:
```

Comment:
Could we just move the file instead of read/write to the new location?

Reply:
I would need to convert it into OuputFile which IMO requires a new function in common for these types of cases (a tool outputs a file and we need that file available). I wasn't able to solve this. Any suggestions?

Reply:
Yeah, you are right. All good with the current solution